### PR TITLE
Swift: fix hidden AST getters

### PR DESCRIPTION
**Pull request by redsun82**
For consistency with the C/C++ QL library, getters of AST elements within the hidden AST should not themselves skip other hidden AST elements.

### CODE REVIEW

File: swift/ql/test/library-tests/hidden-ast/Conversions.expected [hunk 1]
```diff
@@ -0,0 +1,3 @@
+| chained_conversion.swift:2:12:2:14 | (Int) ... | LoadExpr | chained_conversion.swift:2:12:2:14 | .self | DotSelfExpr |
+| chained_conversion.swift:2:12:2:14 | (Int?) ... | InjectIntoOptionalExpr | chained_conversion.swift:2:12:2:14 | (Int) ... | LoadExpr |
+| chained_conversion.swift:2:12:2:14 | .self | DotSelfExpr | chained_conversion.swift:2:12:2:12 | x | DeclRefExpr |
```
Comment (geoffw0) [inline swift/ql/test/library-tests/hidden-ast/Conversions.expected:3]: How would this have been different before the change? (I'm not very familiar with the hidden AST / which elements are hidden)

File: swift/ql/lib/codeql/swift/generated/AvailabilityInfo.qll [hunk 2]
```diff
@@ -46,7 +46,12 @@ module Generated {
     /**
      * Gets the `index`th spec of this availability info (0-based).
      */
-    final AvailabilitySpec getSpec(int index) { result = this.getImmediateSpec(index).resolve() }
+    final AvailabilitySpec getSpec(int index) {
+      exists(AvailabilitySpec immediate |
+        immediate = this.getImmediateSpec(index) and
+        if exists(this.getResolveStep()) then result = immediate else result = immediate.resolve()
+      )
```
Comment (geoffw0) [inline swift/ql/lib/codeql/swift/generated/AvailabilityInfo.qll:53]: This additional and not-quite-trivial QL is repeated many, many times in the diff.  I suppose at code generation time we don't know whether `this.getResolveStep()` will exist or not, so we can't just generate one or the other case.  Still, I wonder if there's a way we can simplify / take out any common code here?

**Review by geoffw0 — COMMENTED** [review]
A couple of things I'd like to discuss.

Definitely want to see a DCA run to be confident there are no unintended consequences here.

**Review by d10c — APPROVED** [review]
LGTM! I'm in favour of merging this before the release, and then changing the modifiers separately. Can we still do a DCA run first?

File: swift/ql/test/library-tests/hidden-ast/Conversions.expected (hunk 1 above)
Comment (d10c) [reply swift/ql/test/library-tests/hidden-ast/Conversions.expected:3]: Before the change, they would all have returned the fully resolved `x` as their sub-expr. Now, you get the full chain without going through the Immediate accessors.

File: swift/ql/lib/codeql/swift/generated/AvailabilityInfo.qll (hunk 2 above)
Comment (redsun82) [reply swift/ql/lib/codeql/swift/generated/AvailabilityInfo.qll:53]: I think one way to limit that is the other thing I propose doing, that is limit this kind of machinery to when the return type of the getter is an `Expr` or a `Pattern`. But not something to be tackled in this PR I think.

File: swift/ql/test/library-tests/hidden-ast/Conversions.expected (hunk 1 above)
Comment (redsun82) [reply swift/ql/test/library-tests/hidden-ast/Conversions.expected:3]: Exactly.

File: swift/ql/lib/codeql/swift/generated/AvailabilityInfo.qll (hunk 2 above)
Comment (geoffw0) [reply swift/ql/lib/codeql/swift/generated/AvailabilityInfo.qll:53]: :+1:

As long as the DCA run looks good performance-wise, I'm OK that we wait for that.

**Comment by redsun82** [comment]
DCA looks good :+1:

**Pull request merged**